### PR TITLE
Add Windows support to main

### PR DESCRIPTION
**Pull request by tschoonj**
Very similar to #31, but now targeting `main`

- [x] Bump to latest version
- [x] Add Windows support
- [x] Add myself as maintainer 

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by tschoonj** [comment]
Ready for review

File: recipe/build_libsigcpp.sh [hunk 1]
```diff
@@ -1,20 +1,8 @@
...
-./configure --prefix="${PREFIX}" \
-  || { cat config.log; exit 1; }
-
-make -j${CPU_COUNT}
-if [[ "${CONDA_BUILD_CROSS_COMPILATION:-}" != "1" || "${CROSSCOMPILING_EMULATOR}" != "" ]]; then
-  make check
-fi
-make install
+ninja -C builddir -j${CPU_COUNT}
+ninja -C builddir test
```
Comment (isuruf) [inline recipe/build_libsigcpp.sh:7]: What happens here when cross compiling and there is no emulator?

Comment (tschoonj) [reply recipe/build_libsigcpp.sh:7]: It would fail I guess. I will remove the line.

Comment (isuruf) [reply recipe/build_libsigcpp.sh:7]: What I wanted to say was that I was surprised that testing passed on cross compiling platform osx-arm64. Turns out that the tests are skipped if we are cross compiling. So, it's okay to have this here. Can you bring this line back?

Comment (tschoonj) [reply recipe/build_libsigcpp.sh:7]: Done

**Pull request merged**